core/trino-spi/src/main/java/io/trino/spi/ptf/TableFunctionSplitProcessor.java

```diff
@@ -27,9 +29,12 @@
     /**
      * This method processes a split. It is called multiple times until the whole output for the split is produced.
      *
-     * @param split a {@link ConnectorSplit} representing a subtask.
+     * @param split a {@link ConnectorSplit} representing a subtask, or {@code null} if a split has already started to be processed,
+     * and the implementation returned a {@link TableFunctionProcessorState.Processed} with
+     * {@link TableFunctionProcessorState.Processed#isUsedInput()} being {@code true}.
```

Comment:
hmm - this is interesting contract :)

Reply:
Yup, the states were primarily designed for the `TableFunctionDataProcessor`.
That processor gets one portion of data at a time, and it declares `isUsedInput()` when the portion is ingested. When it has ingested all the due data,  it gets `null` until it's finished.
In case of the `TableFunctionSplitProcessor`, instead of input data, we have a Split. It is presented to the Processor as a single portion of data. If the processor declares `isUsedInput()`, it gets `null` in subsequent calls until it's finished.